src/promptflow-devkit/promptflow/_proxy/_python_executor_proxy.py

```diff
@@ -73,12 +73,18 @@ async def create(
             )
         return cls(flow_executor)
 
+    @property
+    def has_aggregation(self) -> bool:
+        return self._flow_executor.has_aggregation_node
+
     async def exec_aggregation_async(
         self,
         batch_inputs: Mapping[str, Any],
         aggregation_inputs: Mapping[str, Any],
         run_id: Optional[str] = None,
     ) -> AggregationResult:
+        if isinstance(self._flow_executor, ScriptExecutor):
+            return self._flow_executor._exec_aggregation(aggregation_inputs, run_id=run_id)
         return self._flow_executor.exec_aggregation(batch_inputs, aggregation_inputs, run_id=run_id)
```

Comment:
It is not a good idea to have `if else` here, I think we could implement`exec_aggregation` with the same interface, just ignore batch_inputs in the imlementation.

When we inherit the class, we are going to implement the same interface to avoid such if/else

Reply:
The best practice is to avoid any one usage of `if is_eager_flow` or `if is ScriptExecutor`, but I know it might be impossible now. Then we try to avoid unnecessary usage.

Reply:
Refined.